### PR TITLE
7.2 Developer | ConfigurationFieldOptionsProvider | LRDOCS-7079

### PR DESCRIPTION
**Pull request by rbohl**
https://issues.liferay.com/browse/LRDOCS-7079, draft courtesy of @mwilliams2014. When is this supposed to be available for customers? Via a 7.2 Fix Pack? I didn't find the interface in the 7.2.0-ga1 tag.

### CODE REVIEW

File: developer/frameworks/articles/configuration/11-dynamically-populating-select-fields.markdown [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+[TOC levels=1-4]
+
+Configuration options can be presented in a select list, directly entering each
+label and value in the `@Meta.AD` annotation in the [Configuration
+interface](/docs/7-2/frameworks/-/knowledge_base/f/creating-a-configuration-interface).
+
+```java
+@Meta.AD(
+    deflt = "enabled-with-warning", name = "csv-export",
+    optionLabels = {"Enabled", "enabled-with-warning", "Disabled"},
```
Comment (pei-jung) [inline developer/frameworks/articles/configuration/11-dynamically-populating-select-fields.markdown]: Usually a language keys is used for the label, so the value gets localized. And language keys are usually lower-cased.

File: developer/frameworks/articles/configuration/11-dynamically-populating-select-fields.markdown [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+3.  Use the `getOptions()` method to retrieve the options from the Configuration 
+    interface and populate the configuration fields with their labels and 
+    values, as shown in the example below:
+
+    <!-- Drew or Pei-Jung, we need to describe this code, or provide more
+    realistic example code if we want to. Whatever will best help third party
+    devs see what they can do with the interface. -->
+    ```java    
+    public List<Option> getOptions() {
```
Comment (pei-jung) [inline developer/frameworks/articles/configuration/11-dynamically-populating-select-fields.markdown:69]: The `getOptions` method returns a list of `Option`, and an `Option` object consists of the label and value fields. The labels will be translate to `optionLabels `, and values will be `optionValues` in the config interface. What this code does is  it first gets a list of `AssetRendererFactory` objects, iterate through the list, and populate a new list of `Option` from the list of `AssetRendererFactory`, using the asset's type name as the label and the class name as value.

I think it'd be helpful to reference a sample implementation (ie. `EnabledClassNamesConfigurationFieldOptionsProvider`) and include a screenshot of the configuration to illustrate what the result looks like.

**Comment by pei-jung** [comment]
AFAIK, the interface has already been backported to 7.2, but I'm not sure when this new interface will be available to customer. I guess just track https://issues.liferay.com/browse/LPS-97123 and see when it gets included in a fix pack.

File: developer/frameworks/articles/configuration/11-dynamically-populating-select-fields.markdown (hunk 1 above)
Comment (rbohl) [reply developer/frameworks/articles/configuration/11-dynamically-populating-select-fields.markdown]: Good point. I copied this directly from our source code :scream: 
https://github.com/liferay/liferay-portal/blob/b51b7070c398d44462b282c69324b2bb62050dc4/modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/configuration/DDLWebConfiguration.java#L38-L44

Comment (rbohl) [reply developer/frameworks/articles/configuration/11-dynamically-populating-select-fields.markdown]: Okay, it's fixed in master and on 7.2.x :+1:
https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/configuration/DDLWebConfiguration.java#L38-L44

**Comment by rbohl** [comment]
Hey Pei-Jung can you check this last commit and tell me if this is ready to go? 
https://github.com/pei-jung/liferay-docs/pull/2/commits/d0a216482484f88af442e093347cd357a061df2b

**Comment by rbohl** [comment]
Thanks!
Forwarded: https://github.com/sez11a/liferay-docs/pull/4444

**Pull request closed without merging**